### PR TITLE
Add kill tasks to dcos/marathon.py; Documented in Marathon API here: …

### PR DESCRIPTION
**Pull request by sargun**
…https://mesosphere.github.io/marathon/docs/rest-api.html#delete-v2-apps-appid-tasks

### CODE REVIEW

**Comment by tamarrow** [comment]
@sargun can you add a test please?

**Comment by jsancio** [comment]
@sargun Does this implement DCOS-2941? cc @brndnmtthws

File: cli/dcoscli/marathon/main.py [hunk 1]
```diff
@@ -714,6 +720,32 @@ def _restart(app_id, force):
...
+def _kill(app_id, scale, host):
+    """
+    :param: scale: Scale the app down
+    :type: scale: bool
+    :param: host: Kill only those tasks running on host host
+    :type: string
+    :param app_id: the id of the application
+    :type app_id: str
+    :returns: process return code
+    :rtype: int
```
Comment (tamarrow) [inline cli/dcoscli/marathon/main.py]: can you reorder these in the order of the params (app_id, scale, host) please?

File: cli/tests/integrations/test_marathon.py [hunk 2]
```diff
@@ -754,7 +835,8 @@ def _list_tasks(expected_count, app_id=None):
     result = json.loads(stdout.decode('utf-8'))
 
     assert returncode == 0
-    assert len(result) == expected_count
+    if expected_count >= 0:
+        assert len(result) == expected_count
```
Comment (tamarrow) [inline cli/tests/integrations/test_marathon.py]: when would expected_count be < 0?

File: cli/tests/integrations/test_marathon.py [hunk 3]
```diff
@@ -465,6 +466,86 @@ def test_restarting_app():
...
+                                if task['host'] != kill_host])
+        assert len(not_to_be_killed) > 0
+        assert len(expected_to_be_killed) > 0
+        command = ['dcos',
+                   'marathon',
+                   'app',
+                   'kill',
+                   '--host',
+                   kill_host,
+                   'zero-instance-app']
```
Comment (tamarrow) [inline cli/tests/integrations/test_marathon.py]: can we put these on the same line (as much as we can with the 80char limit)? it makes it a little easier to read.

File: cli/dcoscli/marathon/main.py [hunk 4]
```diff
@@ -714,6 +720,32 @@ def _restart(app_id, force):
...
+    """
+    client = marathon.create_client()
+
+    payload = client.kill_tasks(app_id, host=host, scale=scale)
+    # If scale is provided, the API return a "deploymentResult"
+    # https://github.com/mesosphere/marathon/blob/50366c8/src/main/scala/mesosphere/marathon/api/RestResource.scala#L34-L36
+    if scale:
+        emitter.publish("Started Deployment: {}".format(payload))
```
Comment (tamarrow) [inline cli/dcoscli/marathon/main.py]: deployment should be lowercase

File: cli/dcoscli/marathon/main.py [hunk 5]
```diff
@@ -11,6 +11,7 @@
     dcos marathon app show [--app-version=<app-version>] <app-id>
     dcos marathon app start [--force] <app-id> [<instances>]
     dcos marathon app stop [--force] <app-id>
+    dcos marathon app kill [--scale] [--host=<host>] <app-id>
     dcos marathon app update [--force] <app-id> [<properties>...]
     dcos marathon app version list [--max-count=<max-count>] <app-id>
     dcos marathon deployment list [--json <app-id>]
```
Comment (tamarrow) [inline cli/dcoscli/marathon/main.py]: Can we also add the new arguments to the docstring? https://github.com/mesosphere/dcos-cli/blob/master/cli/dcoscli/marathon/main.py#L30-L101

File: cli/tests/integrations/test_marathon.py [hunk 6]
```diff
@@ -30,6 +30,7 @@ def test_help():
     dcos marathon app show [--app-version=<app-version>] <app-id>
     dcos marathon app start [--force] <app-id> [<instances>]
     dcos marathon app stop [--force] <app-id>
+    dcos marathon app kill [--scale] [--host=<host>] <app-id>
     dcos marathon app update [--force] <app-id> [<properties>...]
```
Comment (tamarrow) [inline cli/tests/integrations/test_marathon.py]: can we add tests for the negative cases also? app-id doesnt exist, or no tasks?

File: cli/dcoscli/marathon/main.py [hunk 7]
```diff
@@ -714,6 +720,32 @@ def _restart(app_id, force):
     return 0
 
 
+def _kill(app_id, scale, host):
+    """
+    :param: scale: Scale the app down
+    :type: scale: bool
+    :param: host: Kill only those tasks running on host host
```
Comment (tamarrow) [inline cli/dcoscli/marathon/main.py]: on host specified

**Comment by tamarrow** [comment]
@sargun looks good, just a few comments. When you're done can you squash your commits and add DCOS-2941 to the commit name?

**Comment by sargun** [comment]
@tamarrow Done.

File: cli/tests/integrations/test_marathon.py (hunk 6 above)
Comment (sargun) [inline cli/tests/integrations/test_marathon.py]: No tasks already is there: test_kill_stopped_app. I added a test for missing apps: test_kill_missing_app

File: cli/dcoscli/marathon/main.py [hunk 8]
```diff
@@ -99,6 +104,8 @@
     <task-id>                   The task id
 
     <scale-factor>              The factor to scale an application group by
+
+    <host>                      The host name to isolate your command to
```
Comment (jsancio) [inline cli/dcoscli/marathon/main.py]: This should go in the `Options:` section as `--host=<host>`. It is not a positional argument.

**Comment by jsancio** [comment]
@sargun Looks good to me. One small comment regarding `--help`.

**Comment by sargun** [comment]
@jsancio I fixed the issue you pointed out. The test_node failure seems to be unrelated to these changes.

**Comment by jsancio** [comment]
Okay. I'll run the build again...

**Comment by jsancio** [comment]
Actually, I think I found the issue. #372 should fix the build.

**Comment by sargun** [comment]
@jsancio Should I merge 372 into my branch, or can you just merge this into master?

**Comment by jsancio** [comment]
@sargun I'll merge it today when we merge #372. We want to make sure that we don't merge anything that doesn't pass even if it because of a bug in the master branch.
